### DISCUSSION TITLE
Setup Apache Reverse Proxy For Code-Server

### DISCUSSION THREAD

**Discussion by FireWtap** [coder/code-server#4262] [Q&A]
I recently set-up code-server on my ubuntu VPS.
If I open it from the browser on my PC from VPS_IP:PORT it works perfectly fine.
However, I recently got an Ipad and I was looking to also use it for coding on-the-go.
After I figured out I need a signed HTTPS certificate in order to use it on IPAD, i set up a reverse proxy with apache from a domain of mine to codeserver

That's the config of the apache website

```
<IfModule mod_ssl.c>
<VirtualHost *:443>
        ServerName MY_DOMAIN
        ServerAdmin webmaster@example.com
        ErrorLog ${APACHE_LOG_DIR}/error.log
        CustomLog ${APACHE_LOG_DIR}/access.log combined

        ProxyRequests Off
        <Proxy *>
          Order deny,allow
          Allow from all
        </Proxy>

        ProxyPass / http://127.0.0.1:1239/
        ProxyPassReverse / http://127.0.0.1:1239/

        <Location />
          Order allow,deny
          Allow from all
        </Location>


Include /etc/letsencrypt/options-ssl-apache.conf
SSLCertificateFile /etc/letsencrypt/live/MY_DOMAIN-0001/fullchain.pem
SSLCertificateKeyFile /etc/letsencrypt/live/MY_DOMAIN-0001/privkey.pem
</VirtualHost>
</IfModule>
```

Now, after I start code-server through the terminal using `code-server --bind-addr 0.0.0.0:1239 ./`, I can connect to my domain and see the login page and the HTTPS working fine.
After the login I only see a white screen and tons of errors in the console stating VSCode failed to initialize.

Any suggestion on how to fix this?
P.S. I get this error only on the reverse proxy, both from the Ipad and my PC. The direct URL still works from my PC but gives white screen on the IPAD due to the lack of the certificate.

**Comment by jsjoeio** [comment]
Hmm...I don't think you'll be able to get around this on iPad. I believe you need valid certificates to get it working. At least that's what I'm aware of.

You could try running `code-server --link` and accessing code-server via that URL on your iPad.

**Comment by FireWtap** [comment]
As I said, i generated the certificate through let's encrypt. :/

**Reply by jsjoeio** [reply]
Hmm you could try this: https://support.apple.com/en-us/HT204477
